internal/pkg/template/templates/workloads/partials/cf/workload-container.yml

```diff
@@ -29,6 +29,16 @@
 {{- if eq .WorkloadType "Load Balanced Web Service"}}
   PortMappings:
     - ContainerPort: !Ref ContainerPort
+{{- if .NLB}}
+  {{if ne .NLB.Listener.TargetPort .NLB.MainContainerPort}} {{/*No need to add additional port if the target port is the same as image port*/}}
```

Comment:
am I understanding this right `.NLB.MainContainerPort` is equal to `s.manifest.ImageConfig.Port`? If so do we need `.NLB.MainContainerPort`?

Reply:
It is equal, but `s.manifest.ImageConfig.Port` is only passed to the parameter `ContainerPort`, and not to `template.Workload`. So basically we need to pass it to the template data somehow. I put it under the `NLB` struct only because it is only needed when `nlb` is enabled.

Reply:
Wouldn't it look confusing to user, having this parameter?

Reply:
hmm either works for me, but it is actually only a condition not a parameter.

Reply:
ya right typo!

I just have general concern in adding stuff to the template vs. changing how the template is rendered. This works, I'm just not sure if it's preferrable

Reply:
If we add an additional param then we need to make sure they are always synced up with each other, which is kinda annoying. That's not a problem now but let's say in the future if we allow them to reuse the template and pass in whatever values to some params, then it would potentially be a problem (though we could fix later as well).

But I agree with you: adding an additional condition does not sound to be an elegant solution to me as well.